java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java

```diff
@@ -83,17 +85,24 @@
   private final boolean[] fileIncluded;
   private final long rowIndexStride;
   private long rowInStripe = 0;
+  private long followRowInStripe = 0;
   private int currentStripe = -1;
   private long rowBaseInStripe = 0;
   private long rowCountInStripe = 0;
   private final BatchReader reader;
   private final OrcIndex indexes;
+  // identifies the columns requiring row indexes
+  private final boolean[] rowIndexCols;
   private final SargApplier sargApp;
   // an array about which row groups aren't skipped
   private boolean[] includedRowGroups = null;
   private final DataReader dataReader;
   private final int maxDiskRangeChunkLimit;
   private final StripePlanner planner;
+  // identifies the type of read: FULL (no filters), LEAD (filters are present)
+  private final ReadLevel readLevel;
```

Comment:
FOLLOW is never set at the RecordReader level. It will only be FULL or LEAD. If this is LEAD then as part of read FOLLOW is implicitly used.

I can clarify further in the comment if that helps.

Reply:
I think I'd remove this and just check whether the row level filter is null or not.